src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Provides extension methods for tracking events with TargetingContext.
+    /// </summary>
+    public static class ApplicationInsightsTelemetryExtensions
+    {
+        /// <summary>
+        /// Extension method to track an event with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackEvent(this TelemetryClient telemetryClient, string eventName, TargetingContext targetingContext, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
+        {
+            if (properties == null)
+            {
+                properties = new Dictionary<string, string>();
+            }
+
+            properties["TargetingId"] = targetingContext.UserId;
+
+            telemetryClient.TrackEvent(eventName, properties, metrics);
+        }
+
+        /// <summary>
+        /// Extension method to track an <see cref="EventTelemetry"/> with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackEvent(this TelemetryClient telemetryClient, EventTelemetry telemetry, TargetingContext targetingContext)
+        {
+            if (telemetry == null)
+            {
+                telemetry = new EventTelemetry();
+            }
+
+            telemetry.Properties["TargetingId"] = targetingContext.UserId;
+
+            telemetryClient.TrackEvent(telemetry);
+        }
+
+        /// <summary>
+        /// Extension method to track a metric with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackMetric(this TelemetryClient telemetryClient, string name, double value, TargetingContext targetingContext, IDictionary<string, string> properties = null)
```

Comment:
At the moment, I believe only Event is in scope.

Reply:
That is true for the whole system, but I think FeatureManagement should support this as well.

Reply:
There's two things that come to mind.

* we can always add we can never take away.
* Metrics are usually aggregated so I see complications when they are used in conjunction with targeting id.

Reply:
Valid points. Removed.